### PR TITLE
Move content-build Nightwatch Daily Accessibility Scan to GHA

### PR DESCRIPTION
**Pull request by timwright12**
## Description

The content-build repository does not currently have a GitHub Actions workflow that executes the daily accessibility scan and reports the results to the #-daily-accessibility-scan channel in Slack. We need to move the daily scan out of Jenkins and into GHA.

Issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/25568

## Testing Done

Successfully completed scans and slack alerts. Will complete tests if the cron executes in master

### CODE REVIEW

File: .github/workflows/daily-accessibility-scan.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
+name: Daily Accessibility Scan
+
+on:
+  schedule:
+    - cron: '0 12 * * 1-5'
```
Comment (timwright12) [inline .github/workflows/daily-accessibility-scan.yml:5]: Hopefully 8am ET mon-fri

File: config/nightwatch.js [hunk 2]
```diff
@@ -85,7 +85,7 @@ module.exports = {
     },
     bestpractice: {
       globals: {
-        rules: ['section508', 'wcag2a', 'wcag2aa', 'best-practice'],
+        rules: ['section508', 'wcag2a', 'wcag2aa'],
```
Comment (timwright12) [inline config/nightwatch.js:88]: Updated per conversation with @1Copenut

File: script/run-nightwatch.sh [hunk 3]
```diff
@@ -24,9 +24,6 @@ done
 # a mock API server as well as a webserver with the content to provide
 # endpoints that selenium can run against.
 
-# Ensure all running servers are terminated on script exit.
-trap 'if [ $(jobs -p) ] ; then kill $(jobs -p); fi' EXIT
```
Comment (timwright12) [inline script/run-nightwatch.sh:28]: This seems like it kept killing off my tests

File: .github/workflows/daily-accessibility-scan.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    name: Run Scan
+    runs-on: self-hosted
+    defaults:
+      run:
+        working-directory: content-build
+
+    env:
+      NODE_EXTRA_CA_CERTS: /etc/ssl/certs/VA-Internal-S2-RCA1-v1.cer.pem
+
+    strategy:
```
Comment (oseasmoran73) [inline .github/workflows/daily-accessibility-scan.yml:23]: Is this still relevant? I don't see an actual strategy associated with matrix

**Review by theodur — APPROVED** [review]

File: .github/workflows/daily-accessibility-scan.yml [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  schedule:
+    - cron: '0 12 * * 1-5'
+
+env:
+  CHROMEDRIVER_FILEPATH: /usr/local/share/chrome_driver/chromedriver
+  CHANNEL_ID: C024DUC9S1K
+
+jobs:
+
+  build:
```
Comment (oseasmoran73) [inline .github/workflows/daily-accessibility-scan.yml:13]:
```suggestion
  accessibility-scan:
```

File: .github/workflows/daily-accessibility-scan.yml [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    name: Run Scan
```
Comment (oseasmoran73) [inline .github/workflows/daily-accessibility-scan.yml:14]:
```suggestion
    name: Accessibility Scan
```

File: .github/workflows/daily-accessibility-scan.yml [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+      - name: Build vagovprod
+        run: NODE_ENV=production npm --no-color run build -- --verbose --buildtype=vagovprod --asset-source=local --drupal-address=https://cms-vets-website-branch-builds-lo9uhqj18nwixunsjadvjsynuni7kk1u.ci.cms.va.gov --pull-drupal --no-drupal-proxy
+        timeout-minutes: 60
+        env:
+          NODE_ENV: production
+
+      - name: Run Accessibility Scan
+        id: accessibility-scan
+        run: WEB_PORT=3002 BUILDTYPE=vagovprod yarn test:accessibility
+        timeout-minutes: 600
```
Comment (oseasmoran73) [inline .github/workflows/daily-accessibility-scan.yml:114]: Seems like overkill, is that how it is currently?

File: .github/workflows/daily-accessibility-scan.yml [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+      - name: Build vagovprod
+        run: NODE_ENV=production npm --no-color run build -- --verbose --buildtype=vagovprod --asset-source=local --drupal-address=https://cms-vets-website-branch-builds-lo9uhqj18nwixunsjadvjsynuni7kk1u.ci.cms.va.gov --pull-drupal --no-drupal-proxy
+        timeout-minutes: 60
+        env:
+          NODE_ENV: production
+
+      - name: Run Accessibility Scan
+        id: accessibility-scan
+        run: WEB_PORT=3002 BUILDTYPE=vagovprod yarn test:accessibility
```
Comment (ddzz) [inline .github/workflows/daily-accessibility-scan.yml:113]: We could move these two variables under `env:`.

**Review by oseasmoran73 — APPROVED** [review]

File: .github/workflows/daily-accessibility-scan.yml [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+    runs-on: self-hosted
+    defaults:
+      run:
+        working-directory: content-build
+
+    env:
+      NODE_EXTRA_CA_CERTS: /etc/ssl/certs/VA-Internal-S2-RCA1-v1.cer.pem
+
+    strategy:
+      fail-fast: true
```
Comment (ddzz) [inline .github/workflows/daily-accessibility-scan.yml:24]: Is `fail-fast` needed here? We're not using the `matrix` strategy.

File: .github/workflows/daily-accessibility-scan.yml (hunk 7 above)
Comment (timwright12) [reply .github/workflows/daily-accessibility-scan.yml:114]: No, I just don't want the test to hang there forever

File: .github/workflows/daily-accessibility-scan.yml [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+            node_modules
+          key: ${{ steps.get-node-version.outputs.NODE_VERSION }}-${{ hashFiles('yarn.lock') }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --prefer-offline --ignore-optional
+        env:
+          YARN_CACHE_FOLDER: ~/.cache/yarn
+
+      - name: Build vagovprod
+        run: NODE_ENV=production npm --no-color run build -- --verbose --buildtype=vagovprod --asset-source=local --drupal-address=https://cms-vets-website-branch-builds-lo9uhqj18nwixunsjadvjsynuni7kk1u.ci.cms.va.gov --pull-drupal --no-drupal-proxy
```
Comment (U-DON) [inline .github/workflows/daily-accessibility-scan.yml:106]:
```suggestion
        run: npm --no-color run build -- --verbose --buildtype=vagovprod --asset-source=local --drupal-address=https://cms-vets-website-branch-builds-lo9uhqj18nwixunsjadvjsynuni7kk1u.ci.cms.va.gov --pull-drupal --no-drupal-proxy
```

Already covered by `env.NODE_ENV`.

File: .github/workflows/daily-accessibility-scan.yml (hunk 8 above)
Comment (timwright12) [reply .github/workflows/daily-accessibility-scan.yml:113]: I actually tried that and it threw errors, not sure why

**Review by U-DON — APPROVED** [review]

File: .github/workflows/daily-accessibility-scan.yml (hunk 9 above)
Comment (timwright12) [reply .github/workflows/daily-accessibility-scan.yml:24]: removed

File: .github/workflows/daily-accessibility-scan.yml [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+      - name: Checkout VSP actions
+        uses: actions/checkout@v2
+        with:
+          repository: department-of-veterans-affairs/vsp-github-actions
+          ref: refs/heads/main
+          token: ${{ env.VA_VSP_BOT_GITHUB_TOKEN }}
+          persist-credentials: false
+          path: ./.github/actions/vsp-github-actions
+
+      - name: Notify Slack
```
Comment (ddzz) [inline .github/workflows/daily-accessibility-scan.yml:58]: Should this be the very first step? Or maybe have `if: ${{ always() }}` here? That would ensure the notification always goes out.

Comment (timwright12) [reply .github/workflows/daily-accessibility-scan.yml:58]: added always() here

File: .github/workflows/daily-accessibility-scan.yml (hunk 10 above)
Comment (timwright12) [reply .github/workflows/daily-accessibility-scan.yml]: added this locally

File: config/nightwatch.js (hunk 2 above)
Comment (1Copenut) [reply config/nightwatch.js:88]: I'd like to get us to best practices, but there's just enough heading levels nested improperly and other nitpicky things floating around that it'd cause unnecessary breaks just now.

**Pull request merged**